dotnet/Cucumber.HtmlFormatterTest/MessagesToHtmlWriterTest.cs

```diff
@@ -31,13 +38,30 @@ public void ItWritesOneMessageToHtml()
                 $"Expected html to contain a testRunStarted message, but instead html contained: {html}");
         }
 
+        [TestMethod]
+        public async Task ItWritesOneMessageToHtmlAsync()
+        {
+            DateTime timestamp = DateTime.UnixEpoch.AddSeconds(10).ToUniversalTime();
+            Envelope envelope = Envelope.Create(new TestRunStarted(Converters.ToTimestamp(timestamp), null));
+            string html = await RenderAsHtmlAsync(envelope);
+            Assert.IsTrue(html.Contains("window.CUCUMBER_MESSAGES = [{\"testRunStarted\":{\"timestamp\":{\"seconds\":10,\"nanos\":0}}}];"),
+                $"Expected html to contain a testRunStarted message, but instead html contained: {html}");
+        }
+
         [TestMethod]
         public void ItWritesNoMessageToHtml()
         {
             string html = RenderAsHtml();
             Assert.IsTrue(html.Contains("window.CUCUMBER_MESSAGES = [];"));
         }
 
+        [TestMethod]
+        public async Task ItWritesNoMessageToHtmlAsync()
+        {
+            string html = await RenderAsHtmlAsync();
+            Assert.IsTrue(html.Contains("window.CUCUMBER_MESSAGES = [];"));
+        }
+
         [TestMethod]
         public void ItThrowsWhenWritingAfterClose()
```

Comment:
(GitHub didn't allow me to add this comment to line 69)

These tests (testing the sync) should be deleted or wrap the constructor call with pragmas to disable this warning explicitly, like 

```
#pragma warning disable CS0618 // Type or member is obsolete
            MessagesToHtmlWriter messagesToHtmlWriter = new MessagesToHtmlWriter(bytes, serializer);
#pragma warning restore CS0618 // Type or member is obsolete
```

Same for all other warnings in this file.

In general, the compilation should produce no warnings.

Reply:
I've made both sets of changes per your requests.